### DISCUSSION TITLE
I have a problem with the environment variable  module.vue

### DISCUSSION THREAD

**Discussion by kintcamara** [directus/directus#14563] [Feature Requests]
hello, I'm really in need, it's been two days that I try to recover the environment variable in the file extensions/modules/name-module/src/.env from extensions/modules/name-module/module.vue in my docker container. can someone help me please?

**Comment by br41nslug** [comment]
The simple answer is you cannot 😬 Frontend extensions like interfaces,displays,modules etc run in the Vue context in the browser and have no direct awareness of environment variables on the server. What are you trying to achieve with this env file? I'm sure there are alternative ways of getting there 😄

**Reply by kintcamara** [reply]
I use algolia and I put the environment variables in the file in "extensions/modules/nom-module/src/.env" and I want to retrieve these environment variables from the file extensions/modules/nom-module/ src/module.vue with process but it triggers an error

**Reply by rijkvanzanten** [reply]
You'd use a `.env` file for configuration that's private that you don't want to expose to the client. By including it in a `.vue` component (regardless of how you'd achieve that), you're exposing that private data to the client. If the key is public anyways, you might as well just use a variable in the Vue component 🙂

**Reply by kintcamara** [reply]
I understand thank you very much for the answer, I will surely have other questions because I am migrating from directus v8 to v9

**Comment by taco-chen** [comment]
Hi, I met a familiar situation like this question. I want to create a `display` extension to make a slug field into a clickable link with the URL prefix. For instance, it will be `https://example.com/articles/{slug}` in my case. 

![Screenshot 2022-11-11 at 12 25 54 AM](https://user-images.githubusercontent.com/2115142/201151301-e996c723-94b5-4e52-953d-d555585426a9.png)

However, since I have different environments from development, staging to production, I'd like to use `.env` to dynamic load the prefix:

Production:
.env.production: `DOMAIN=https://example.com`
`{process.env.DOMAIN}/articles/{slug}`

Staging
.env.staging: `DOMAIN=https://staging.example.com`
`{process.env.DOMAIN}/articles/{slug}`

Development
.env: `DOMAIN=http://localhost:3000`
`{process.env.DOMAIN}/articles/{slug}`

But I cannot find a way to achieve that. Would you please guide me on what kind of extension can I use? Or any workaround to fix this in the `display` extension?

**Reply by taco-chen** [reply]
I've already tried to use VUE_APP_DOMAIN to expose the `.env` variable (https://stackoverflow.com/questions/50828904/using-environment-variables-with-vue-js, https://cli.vuejs.org/guide/mode-and-env.html, https://dev.to/ratracegrad/how-to-use-environment-variables-in-vue-js-4ko7), but it seems not to work as well.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
